### PR TITLE
Add date param for ranking page

### PR DESCRIPTION
**Pull request by chejennifer**
- when there is a date specified for the ranking page, get data using within-places endpoint and do on the fly ranking calculation
- if map tool has a date set, "explore ranking" link will redirect to ranking page with that specified date

### CODE REVIEW

**Comment by pradh** [comment]
Very nice!

**Review by beets — APPROVED** [review]
thanks for this and the fixes! nice change

File: static/js/ranking/ranking_page.tsx [hunk 1]
```diff
@@ -114,7 +125,7 @@ class Page extends React.Component<RankingPagePropType, RankingPageStateType> {
   private renderToggle(): JSX.Element {
     const svData = this.state.data;
     if (svData.rankAll) return;
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply static/js/ranking/ranking_page.tsx]: done

File: static/js/ranking/ranking_page.tsx [hunk 2]
```diff
@@ -305,6 +330,78 @@ class Page extends React.Component<RankingPagePropType, RankingPageStateType> {
...
+            const popDate = getPopulationDate(popSeries, placeStat);
+            const popValue = popSeries.val[popDate];
+            if (popValue < MIN_POPULATION) {
+              continue;
+            }
+            if (this.props.isPerCapita) {
+              value /= popValue;
+            }
+          } else if (this.props.isPerCapita) {
+            // empty pop series but is per capita
```
[the remark this answers is not in the document]
Comment (chejennifer) [reply static/js/ranking/ranking_page.tsx:379]: done

**Pull request merged**